Twitch/Twitch.Base/Services/NewAPI/ScheduleService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Newtonsoft.Json;
+using StreamingClient.Base.Util;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Twitch.Base.Models.NewAPI;
+using Twitch.Base.Models.NewAPI.Schedule;
+using Twitch.Base.Models.NewAPI.Users;
+
+namespace Twitch.Base.Services.NewAPI
+{
+    /// <summary>
+    /// APIs for schedule-based services
+    /// </summary>
+    public class ScheduleService : NewTwitchAPIServiceBase
+    {
+        /// <summary>
+        /// Creates an instance of the ScheduleService.
+        /// </summary>
+        /// <param name="connection">The Twitch connection to use</param>
+        public ScheduleService(TwitchConnection connection) : base(connection) { }
+
+        /// <summary>
+        /// Get stream schedule for broadcaster
+        /// </summary>
+        /// <param name="broadcaster">broadcaster</param>
+        /// <param name="maxScheduleSegments">Maximum schedule segment results to return</param>
+        /// <returns>Broadcaster's schedule details</returns>
+        public async Task<ScheduleModel> GetSchedule(UserModel broadcaster, int maxScheduleSegments)
+        {
+            Validator.ValidateVariable(broadcaster, "broadcaster");
+            string cursor = null;
+            ScheduleModel scheduleModel = new ScheduleModel();
+            scheduleModel.segments = new List<ScheduleSegmentModel>();
+
+            int totalScheduleSegments = 0;
+            bool firstPass = true;
+            NewTwitchAPISingleDataRestResult results;
+            do
+            {
+                int itemsToRetrieve = maxScheduleSegments - totalScheduleSegments > 25 ? 25 : maxScheduleSegments - totalScheduleSegments;
+                results = await this.GetPagedSingleDataResultAsync("schedule?broadcaster_id=" + broadcaster.id, itemsToRetrieve, cursor);
+                cursor = results.Cursor;
+
+                if (results.data != null)
+                {
+                    if (firstPass)
+                    {
+                        scheduleModel = results.data.ToObject<ScheduleModel>();
+                        firstPass = false;
+                        totalScheduleSegments += results.data.ToObject<ScheduleModel>().segments.Count;
+                    }
+                    else
+                    {
+                        var segments = results.data.ToObject<ScheduleModel>().segments;
```

Comment:
Related the comment directly above this, this else clause them simply becomes:

scheduleModel.segments.AddRange(segments);

where you reference the deserialized result above.

Reply:
> Let me know if you have any questions on the comments. If you're uncertain about the changes, let me know and we can push this through & I can make them on my end.

I was tied up this evening but will look this all over tomorrow and work on changes.  :)